python/ray/dag/compiled_dag_node.py

```diff
@@ -153,6 +153,14 @@ def do_exec_tasks(
         for task in tasks:
             task.prepare(overlap_gpu_communication=overlap_gpu_communication)
 
+        torch_device = ChannelContext.get_current().torch_device
+        if torch_device.type == "cuda":
+            # Compiled Graphs applications (e.g., vLLM) may not properly
+            # set the CUDA device. We explicitly set the device here.
+            import torch
+
+            torch.cuda.set_device(torch_device)
```

Comment:
Let's not set the torch context. It's OK to automatically update Ray's context but in general we should avoid overriding user context for third-party libraries - it can lead to unintuitive behavior/silent bugs for users and it's difficult to opt out.

Instead can we make sure to use the right torch device during stream creation? It looks like we're not using the device from the context right now: https://github.com/ray-project/ray/blob/5896b3fa4c7e07a7d7cfc8f248bd8d50802cbb4c/python/ray/experimental/channel/nccl_group.py#L109

Reply:
Fixed